packages/crypto/src/index.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Generates cryptographically secure random bytes.
+ * @param size The number of random bytes to generate. The `size` must not be greater than 65536.
+ * @returns A Uint8Array containing the random bytes.
+ * @throws {RangeError} If `size` is greater than 65536.
+ */
+export function randomBytes(size: number): Uint8Array {
+  const MAX_BYTES = 65536; // limit of Crypto.getRandomValues()
+  if (size > MAX_BYTES) {
+    throw new RangeError(`size must be less than or equal to ${MAX_BYTES}`);
+  }
+
+  const bytes = new Uint8Array(size);
+  try {
+    crypto.getRandomValues(bytes); // Node.js 20+ and modern browsers
+  } catch {
+    /* eslint-disable @typescript-eslint/no-var-requires */
+    require("crypto").getRandomValues(bytes); // Node.js 18
```

Comment:
Consider using the `exports` field in `package.json` to avoid using the `require` API

Reply:
How exactly ?

It seems `exports` field in `package.json` can not solve this problem.

`require("crypto").getRandomValues(...)` here targets only Node.js 18 because Web Crypto API 

`crypto.getRandomValues()` is not available in `global`

Reply:
sry, I misremembered the [browser](https://github.com/defunctzombie/package-browser-field-spec) keyword as being in `exports`, we should have used the `browser` keyword to export a different `crypto` runtime

Reply:
The problem here is NOT about Browser or Node.js .

The problem here is about Node.js 18 or Node.js 20.

In modern browsers and Node.js 20, `crypto.getRandomValues()` is available in global namespace.

In Node.js 18, `crypto.getRandomValues()` is NOT in the global namespace,
so we have to `require('crypto')` to use it.

Reply:
For Nodejs-compatible runtimes, directly using the `require('crypto')` is fine. However, for a browser runtime or bundler, it doesn't always seem to work properly when using `require('package')` https://github.com/nervosnetwork/neuron/pull/3148

```js
// node.js

export { randomBytes } from 'crypto'

```

```js
// browser
export function randomBytes(size: number): Uint8Array {
  // ...
}
```

Reply:
Added `crypto: false` in `browser` field in `package.json` to avoid issue like nervosnetwork/neuron#3148
```json
  "browser": {
    "crypto": false
  },
```
I've created a npm package https://www.npmjs.com/package/nocrypto to test the `browser` field behavior against webpack 5 and it works.

Browsers support `crypto.getRandomValues()` ten year ago so we don't have to worry about browsers.
https://developer.mozilla.org/en-US/docs/Web/API/Web_Crypto_API#browser_compatibility

For Node.js 18, `require('crypto')` is still necessary for now, but after May 2025, we can drop support for Node 18 and remove the `try {} catch {}` and `require('crypto')` .

We don't want code below because Node's `crypto.randomBytes` returns a `Buffer`.
```ts
// node.js
export { randomBytes } from 'crypto'
```

Reply:
Updated to use `randomBytes` from `@noble/hashes` package